pkg/migration-tool/planner/slab.go

```diff
@@ -118,6 +144,48 @@ func (s *Slab) Fetch(ctx context.Context, client *utils.Client, mint, maxt int64
 	return nil
 }
 
+func (s *Slab) mergeSubSlabsToSlab(subSlabs []*utils.PrompbResponse) []*prompb.TimeSeries {
+	l := len(subSlabs)
+	s.UpdatePBarMax(s.PBarMax() + 2)
+	s.SetDescription(fmt.Sprintf("combining fetched series from %d responses", l), 1)
+	timeseries := make(map[string]*prompb.TimeSeries)
+	for i := 0; i < l; i++ {
+		ts := subSlabs[i].Result.Timeseries
+		for _, series := range ts {
+			if s, ok := timeseries[getLabelsStr(series.Labels)]; len(series.Samples) > 0 && ok {
```

Comment:
I would have a variable for  getLabelsStr(series.Labels) as that's an expensive function you don't want to call multiple time

Reply:
Do you mean as a cache? Or just for readability?

Reply:
Lets discuss this on meet

cc @cevian